skglm/tests/test_prox_newton.py

```diff
@@ -37,14 +38,19 @@ def test_pn_vs_sklearn(rho, X_density):
     alpha = rho * alpha_max
 
     sk_log_reg = LogisticRegression(penalty='l1', C=1/(n_samples * alpha),
-                                    fit_intercept=False, tol=1e-9, solver='liblinear')
+                                    fit_intercept=fit_intercept, random_state=0,
+                                    tol=1e-9, solver='saga', max_iter=1_000_000)
     sk_log_reg.fit(X, y)
 
     log_datafit = compiled_clone(Logistic())
     l1_penalty = compiled_clone(L1(alpha))
-    w = ProxNewton(tol=1e-9).solve(X, y, log_datafit, l1_penalty)[0]
+    prox_solver = ProxNewton(fit_intercept=fit_intercept, tol=1e-9)
+    w = prox_solver.solve(X, y, log_datafit, l1_penalty)[0]
 
-    np.testing.assert_allclose(w, sk_log_reg.coef_.flatten(), rtol=1e-6, atol=1e-6)
+    np.testing.assert_allclose(w[:n_features], sk_log_reg.coef_.flatten(),
+                               rtol=1e-4, atol=1e-3)
+    if fit_intercept:
+        np.testing.assert_allclose(w[-1], sk_log_reg.intercept_, rtol=1e-4, atol=1e-3)
```

Comment:
@Badr-MOUFAD use atol and rtol only when needed. The tests seem to pass even when they take their default values here. If not, consider lowering the tol.

atol=1e-3 may be big in some settings, so better avoid it.

Reply:
spoke too fast, some tests seem to break. Can you check if lowering the tolerance of solvers allows us to remove the rtol and atol ? Or just to use atol=1e-9 ?